### PR TITLE
fix(web): slideshow ascending order when video has ended

### PR DESCRIPTION
**Pull request by martabal**
When video has ended, the slideshow order is not respected. This PR fix it.

### CODE REVIEW

File: web/src/lib/components/asset-viewer/video-viewer.svelte [hunk 1]
```diff
@@ -3,22 +3,24 @@
   import { getAssetFileUrl, getAssetThumbnailUrl } from '$lib/utils';
   import { handleError } from '$lib/utils/handle-error';
   import { ThumbnailFormat } from '@immich/sdk';
-  import { createEventDispatcher } from 'svelte';
   import { fade } from 'svelte/transition';
   import LoadingSpinner from '../shared-components/loading-spinner.svelte';
 
   export let assetId: string;
+  export let onVideoStarted: (() => void) | undefined = undefined;
+  export let onVideoEnded: () => void;
```
Comment (danieldietzler) [inline web/src/lib/components/asset-viewer/video-viewer.svelte:11]: Nit but I would probably prefer `onVideoStart` and `onVideoEnd`

**Review by danieldietzler — APPROVED** [review]
LGTM!

File: web/src/lib/components/asset-viewer/video-viewer.svelte [hunk 2]
```diff
@@ -3,22 +3,24 @@
...
-  const dispatch = createEventDispatcher<{ onVideoEnded: void; onVideoStarted: void }>();
 
   const handleCanPlay = async (event: Event) => {
     try {
       const video = event.currentTarget as HTMLVideoElement;
       video.muted = true;
       await video.play();
       video.muted = false;
-      dispatch('onVideoStarted');
+      if (onVideoStart) {
```
[the remark this answers is not in the document]
Comment (danieldietzler) [reply web/src/lib/components/asset-viewer/video-viewer.svelte:21]: You would want to do `onVideoStart && onVideoStart()` and that should work, yea

[the remark this answers is not in the document]
Comment (martabal) [reply web/src/lib/components/asset-viewer/video-viewer.svelte]: Nice !

File: web/src/lib/components/asset-viewer/asset-viewer.svelte [hunk 3]
```diff
@@ -447,7 +447,9 @@
 
   const handleVideoEnded = async () => {
     if ($slideshowState === SlideshowState.PlaySlideshow) {
-      await navigateAsset('next');
+      $slideshowNavigation === SlideshowNavigation.AscendingOrder
+        ? await navigateAsset('previous')
+        : await navigateAsset('next');
```
Comment (jrasm91) [inline web/src/lib/components/asset-viewer/asset-viewer.svelte:452]: Man, I really dislike this. We're adding more duplicate code everywhere and the asset-viewer is even more coupled to the slideshow functionality than it was before. The component should only be in charge of rendering an asset, not managing all this unrelated state.

Comment (martabal) [reply web/src/lib/components/asset-viewer/asset-viewer.svelte]: I moved everything to `navigateAsset`

**Pull request closed without merging**